src/irmin-pack/unix/ext.ml

```diff
@@ -667,9 +667,61 @@ module Maker (Config : Conf.S) = struct
     let stats = Stats.run
     let reload = X.Repo.reload
     let flush = X.Repo.flush
-    let start_gc = X.Repo.Gc.start_exn
-    let finalise_gc = X.Repo.Gc.finalise_exn ?hook:None
-    let finalise_gc_with_hook = X.Repo.Gc.finalise_exn
+
+    module Gc = struct
+      type throttle = [ `Block | `Skip ]
+
+      let catch_errors context error =
+        let backtrace = Printexc.get_backtrace () in
+        let error_msg =
+          match error with
+          | Errors.Pack_error error ->
+              Fmt.str "Pack_error: %a" Errors.pp_base_error error
+          | Irmin.Closed -> "Closed"
+          | Irmin_pack.RO_not_allowed -> "RO_not_allowed"
+          | Unix.Unix_error (err, s1, s2) ->
+              let pp = Irmin.Type.pp Io.misc_error_t in
+              Fmt.str "Unix_error: %a" pp (err, s1, s2)
+          | exn -> raise exn
```

Comment:
Copied from tezos integration. I'd like to improve this but feels better at least having it internal to our code.